core/server/worker/src/test/java/alluxio/worker/block/BlockMasterClientTest.java

```diff
@@ -77,4 +121,314 @@ public void convertBlockListMapToProtoMergeDirsInSameTier() {
+    );
+
+    client.commitBlockInUfs(blockId, length);
+
+    assertEquals(1, committedUfsBlocks.size());
+    assertEquals(length, (long) committedUfsBlocks.get(blockId));
+  }
+
+  @Test
+  public void testGetId() throws Exception {
+    WorkerNetAddress testExistsAddress = new WorkerNetAddress();
+    testExistsAddress.setTieredIdentity(new TieredIdentity(new ArrayList<>()));
+    WorkerNetAddress testNonExistsAddress = new WorkerNetAddress();
+    testNonExistsAddress.setHost("1.2.3.4");
+    long workerId = 0L;
+    Map<WorkerNetAddress, Long> workerIds = ImmutableMap.of(testExistsAddress, workerId);
+
+    Channel mockChannel = createMockService(
+        new BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase() {
+          @Override
+          public void getWorkerId(GetWorkerIdPRequest request,
+                                  StreamObserver<GetWorkerIdPResponse> responseObserver) {
+            WorkerNetAddress address = GrpcUtils.fromProto(request.getWorkerNetAddress());
+            Long id = workerIds.get(address);
+            if (id == null) {
+              id = -1L;
+            }
+            responseObserver.onNext(GetWorkerIdPResponse.newBuilder().setWorkerId(id).build());
+            responseObserver.onCompleted();
+          }
+        });
+
+    BlockMasterClient client = new MockStubBlockMasterClient(
+        MasterClientContext.newBuilder(ClientContext.create(mConf)).build(),
+        mockChannel);
+
+    assertEquals(workerId, client.getId(testExistsAddress));
+    assertEquals(-1L, client.getId(testNonExistsAddress));
+  }
+
+  @Test
+  public void testHeartBeat() throws Exception {
+    final long workerId = 0L;
+    final Map<String, Long> capacityBytesOnTiers = ImmutableMap.of("MEM", 1024 * 1024L);
+    final Map<String, Long> usedBytesOnTiers = ImmutableMap.of("MEM", 1024L);
+    final List<Long> removedBlocks = ImmutableList.of();
+    final Map<BlockStoreLocation, List<Long>> addedBlocks = ImmutableMap.of();
+    final Map<String, List<String>> lostStorage = ImmutableMap.of(
+        "MEM",
+        ImmutableList.of("/tmp/lost")
+    );
+    final List<Metric> metrics = ImmutableList.of();
+
+    Channel mockChannel = createMockService(
+        new BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase() {
+          @Override
+          public void blockHeartbeat(BlockHeartbeatPRequest request,
+                                     StreamObserver<BlockHeartbeatPResponse> responseObserver) {
+            responseObserver.onNext(
+                BlockHeartbeatPResponse.newBuilder().setCommand(Command.newBuilder().setCommandType(
+                    CommandType.Nothing)).build()
+            );
+            responseObserver.onCompleted();
+          }
+        });
+
+    BlockMasterClient client = new MockStubBlockMasterClient(
+        MasterClientContext.newBuilder(ClientContext.create(mConf)).build(),
+        mockChannel
+    );
+
+    assertEquals(CommandType.Nothing, client.heartbeat(
```

Comment:
Also need to verify that the server receives the same capacity and metrics info as the client has sent.

Reply:
Makes sense. I made some verification in the new commit. Does that looks good to you?